emukit/quadrature/methods/warped_bq_model.py

```diff
@@ -111,17 +111,18 @@ def set_data(self, X: np.ndarray, Y: np.ndarray) -> None:
         :param X: Observation locations, shape (n_points, input_dim)
         :param Y: Integrand observations at X, shape (n_points, 1)
         """
-        self.update_parameters(X, Y)
+        self._warping.update_parameters(**self.compute_warping_params(X, Y))
         self.base_gp.set_data(X, self._warping.inverse_transform(Y))
 
-    def update_parameters(self, X: np.ndarray, Y: np.ndarray) -> None:
-        """Update parameters of the model that are not being optimized. Override this method on case parameters
-        are data dependent.
+    def compute_warping_params(self, X: np.ndarray, Y: np.ndarray) -> dict:
+        """Compute parameters of the warping that are dependent on data, and that are not being optimized.
+        Override this method on case parameters are data dependent.
 
         :param X: Observation locations, shape (n_points, input_dim)
         :param Y: Integrand observations at X, shape (n_points, 1)
+        :returns : Dictionary containing new warping parameters. Names of parameters are the keys.
         """
-        pass
+        return {}
```

Comment:
I am a bit out of context here, can you bring me up to speed please? Is this class supposed to be used directly, or only serve as a base? If the latter, maybe consider turning it into an abstract class too.

Also, since we don't have any overrides in this PR, I assume they will be coming later, right?

Reply:
This is the default behavior (empty dict means parameters stay the same). [Here](https://github.com/EmuKit/emukit/pull/375/files#diff-34e3d9470acc26afa739a9cf396ac75ca15dc290975d58ac165e97ed87620e53R57) is an example of it being overridden.

Reply:
I see. Ok, let's leave it as is then, thanks for clarification. I keep getting slightly confused about the structure of our BQ part